aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/options/AWSCognitoAuthUpdateUserAttributesOptions.kt

```diff
@@ -0,0 +1,110 @@
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.auth.cognito.options
+
+import androidx.core.util.ObjectsCompat
+import com.amplifyframework.auth.options.AuthUpdateUserAttributesOptions
+import com.amplifyframework.util.Immutable
+import java.util.Objects.requireNonNull
+/**
+ * Cognito extension of update user attributes options to add the platform specific fields.
+ */
+class AWSCognitoAuthUpdateUserAttributesOptions
+/**
+ * Advanced options for update user attributes.
+ * @param metadata Additional custom attributes to be sent to the service such as information about the client
+ */
+protected constructor(
+    val metadata: Map<String, String>
+) : AuthUpdateUserAttributesOptions() {
+
+    override fun hashCode(): Int {
+        return ObjectsCompat.hash(
+            metadata
+        )
+    }
+
+    override fun equals(obj: Any?): Boolean {
+        return if (this === obj) {
+            true
+        } else if (obj == null || javaClass != obj.javaClass) {
+            false
+        } else {
+            val authUpdateUserAttributesOptions = obj as AWSCognitoAuthUpdateUserAttributesOptions
+            ObjectsCompat.equals(metadata, authUpdateUserAttributesOptions.metadata)
+        }
+    }
+
+    override fun toString(): String {
+        return "AWSCognitoAuthUpdateUserAttributesOptions{" +
+            "metadata=" + metadata +
+            '}'
+    }
+
+    /**
+     * The builder for this class.
+     */
+    class CognitoBuilder : Builder<CognitoBuilder>() {
+        private val metadata: MutableMap<String, String>
+
+        /**
+         * Returns the type of builder this is to support proper flow with it being an extended class.
+         * @return the type of builder this is to support proper flow with it being an extended class.
+         */
+        override fun getThis(): CognitoBuilder {
+            return this
+        }
+
+        /**
+         * Set the metadata field for the object being built.
+         * @param metadata Custom user metadata to be sent with the update user attributes request.
+         * @return The builder object to continue building.
+         */
+        fun metadata(metadata: Map<String, String>): CognitoBuilder {
+            requireNonNull(metadata)
+            this.metadata.clear()
+            this.metadata.putAll(metadata)
+            return getThis()
+        }
+
+        /**
+         * Construct and return the object with the values set in the builder.
+         * @return a new instance of AWSCognitoAuthUpdateUserAttributesOptions with the values specified in the builder.
+         */
+        override fun build(): AuthUpdateUserAttributesOptions {
+            return AWSCognitoAuthUpdateUserAttributesOptions(
+                Immutable.of(metadata)
+            )
+        }
+
+        /**
+         * Constructor for the builder.
+         */
+        init {
+            metadata = HashMap()
```

Comment:
Initialization can be combined with declaration.

Reply:
yes done